test/bundle/x509bundle/test_x509_bundle.py

```diff
@@ -0,0 +1,254 @@
+    x509_bundle = X509Bundle.parse(trust_domain, bundle_bytes)
+
+    bundle_path = tmpdir.join('bundle.pem')
+    X509Bundle.save(x509_bundle, bundle_path, serialization.Encoding.PEM)
+
+    saved_bundle = X509Bundle.load(
+        trust_domain, bundle_path, serialization.Encoding.PEM
+    )
+
+    assert saved_bundle.trust_domain() == trust_domain
+    assert len(saved_bundle.x509_authorities()) == 2
+
+    expected_subjects = ['CN=PEMUTILTEST1', 'CN=PEMUTILTEST2']
+    authority1 = saved_bundle.x509_authorities().pop()
+    assert isinstance(authority1, Certificate)
+    assert authority1.subject.rfc4514_string() in expected_subjects
+
+    authority2 = saved_bundle.x509_authorities().pop()
+    assert isinstance(authority2, Certificate)
+    assert authority2.subject.rfc4514_string() in expected_subjects
+
+
+def test_save_bundle_der_encoded(tmpdir):
+    bundle_bytes = read_bytes('certs.pem')
+    # create the X509Bundle to be saved
+    x509_bundle = X509Bundle.parse(trust_domain, bundle_bytes)
+
+    bundle_path = tmpdir.join('bundle.pem')
+    X509Bundle.save(x509_bundle, bundle_path, serialization.Encoding.DER)
+
+    saved_bundle = X509Bundle.load(
+        trust_domain, bundle_path, serialization.Encoding.DER
+    )
+
+    assert saved_bundle.trust_domain() == trust_domain
+    assert len(saved_bundle.x509_authorities()) == 2
+
+    expected_subjects = ['CN=PEMUTILTEST1', 'CN=PEMUTILTEST2']
+    authority1 = saved_bundle.x509_authorities().pop()
+    assert isinstance(authority1, Certificate)
+    assert authority1.subject.rfc4514_string() in expected_subjects
+
+    authority2 = saved_bundle.x509_authorities().pop()
+    assert isinstance(authority2, Certificate)
+    assert authority2.subject.rfc4514_string() in expected_subjects
+
+
+def test_save_non_supported_encoding(tmpdir):
+    bundle_bytes = read_bytes('certs.pem')
+    # create the X509Bundle to be saved
+    x509_bundle = X509Bundle.parse(trust_domain, bundle_bytes)
+
+    bundle_path = tmpdir.join('bundle.pem')
+
+    with pytest.raises(ValueError) as err:
+        X509Bundle.save(x509_bundle, bundle_path, serialization.Encoding.Raw)
+
+    assert (
+        str(err.value)
+        == 'Encoding not supported: Encoding.Raw. Expected \'PEM\' or \'DER\'.'
+    )
+
+
+def test_add_and_remove_authority():
+    bundle = X509Bundle(trust_domain, None)
+    pem_certs = pem.parse_file(_TEST_CERTS_PATH.format('certs.pem'))
+    x509_cert_1 = x509.load_pem_x509_certificate(
+        pem_certs[0].as_bytes(), default_backend()
+    )
+    x509_cert_2 = x509.load_pem_x509_certificate(
+        pem_certs[1].as_bytes(), default_backend()
+    )
+
+    bundle.add_authority(x509_cert_1)
+    bundle.add_authority(x509_cert_2)
+
+    assert len(bundle.x509_authorities()) == 2
+
+    for a in bundle.x509_authorities():
+        assert isinstance(a, Certificate)
+        assert 'CN=PEMUTILTEST' in a.subject.rfc4514_string()
+
+    bundle.remove_authority(x509_cert_1)
```

Comment:
nit: what about removing an unexistent Certificate and adding one that is already there?

Reply:
Added it to the test: add and remove multiple times the same certificate.